gsa/src/web/entities/filterprovider.js

```diff
@@ -0,0 +1,130 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import React, {useEffect} from 'react';
+
+import {useSelector, useDispatch} from 'react-redux';
+
+import Filter from 'gmp/models/filter';
+
+import {isDefined} from 'gmp/utils/identity';
+
+import Loading from 'web/components/loading/loading';
+
+import getPage from 'web/store/pages/selectors';
+import {loadUserSettingDefault} from 'web/store/usersettings/defaults/actions';
+import {getUserSettingsDefaults} from 'web/store/usersettings/defaults/selectors';
+import {loadUserSettingsDefaultFilter} from 'web/store/usersettings/defaultfilters/actions';
+import {getUserSettingsDefaultFilter} from 'web/store/usersettings/defaultfilters/selectors';
+
+import PropTypes from 'web/utils/proptypes';
+import withGmp from 'web/utils/withGmp';
+
+const ROWS_PER_PAGE_SETTING_ID = '5f5a8712-8017-11e1-8556-406186ea4fc5';
+export const DEFAULT_FALLBACK_FILTER = Filter.fromString('sort=name first=1');
+
+const FilterProvider = ({
+  children,
+  fallbackFilter,
+  gmp,
+  gmpname,
+  locationQuery = {},
+}) => {
+  const dispatch = useDispatch();
+
+  let returnedFilter;
+
+  let defaultSettingFilter = useSelector(state =>
+    getUserSettingsDefaultFilter(state, gmpname).getFilter(),
+  );
+
+  useEffect(() => {
+    if (!isDefined(defaultSettingFilter)) {
+      dispatch(loadUserSettingsDefaultFilter(gmp)(gmpname));
+    }
+  }, [defaultSettingFilter, dispatch, gmp, gmpname]);
+
+  const rowsPerPage = useSelector(state => {
+    const rows = getUserSettingsDefaults(state).getValueByName('rowsperpage');
+    return rows;
+  });
+
+  useEffect(() => {
+    if (!isDefined(rowsPerPage)) {
+      dispatch(loadUserSettingDefault(gmp)(ROWS_PER_PAGE_SETTING_ID));
+    }
+  }, [returnedFilter, rowsPerPage, gmp, dispatch]);
+
+  const pageFilter = useSelector(state => {
+    const pFilter = getPage(state).getFilter(gmpname);
+    return pFilter;
+  });
+
+  if (isDefined(locationQuery.filter)) {
+    returnedFilter = Filter.fromString(locationQuery.filter);
+  } else if (isDefined(pageFilter)) {
+    returnedFilter = pageFilter;
+  } else if (isDefined(defaultSettingFilter) && defaultSettingFilter !== null) {
+    returnedFilter = defaultSettingFilter;
+  } else if (isDefined(fallbackFilter)) {
+    returnedFilter = fallbackFilter;
+  } else {
+    returnedFilter = DEFAULT_FALLBACK_FILTER;
+  }
+
+  // TODO remove the following line once the resultsfilter is working correctly
+  if (gmpname === 'result') {
+    defaultSettingFilter = null;
+  }
+
+  if (!returnedFilter.has('rows') && isDefined(rowsPerPage)) {
+    returnedFilter.set('rows', rowsPerPage);
+  }
+
+  const showChildren =
+    isDefined(returnedFilter) &&
+    isDefined(defaultSettingFilter) &&
+    isDefined(rowsPerPage);
+
+  return (
+    <React.Fragment>
+      {showChildren ? (
+        <React.Fragment>
+          <span data-testid="awaiting-span" />
```

Comment:
Should be dropped and only be rendered in the tests.
```javascript
const renderFunc = jest.fn().mockReturnValue(<span data-testid="awaiting-span" />);
```
```suggestion
```

Reply:
But it's an internal thing in FilterProvider. If I remove it there and put it into the test as a child or so, the test doesn't work anymore.